source/bson-binary-vector/bson-binary-vector.md

```diff
@@ -0,0 +1,103 @@
+
+- Status: Pending
+- Minimum Server Version: N/A
+
+______________________________________________________________________
+
+## Abstract
+
+This document describes the subtype of the Binary BSON type used for efficient storage and retrieval of vectors. Vectors
+here refer to densely packed arrays of numbers, all of the same type.
+
+## Motivation
+
+These representations correspond to the numeric types supported by popular numerical libraries for vector processing,
+such as NumPy, PyTorch, TensorFlow and Apache Arrow. Storing and retrieving vector data using the same densely packed
+format used by these libraries can result in significant memory savings and processing efficiency.
+
+### META
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and
+"OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+## Specification
+
+This specification introduces a new BSON binary subtype, the vector, with value `"\x09"`.
+
+Drivers SHOULD provide idiomatic APIs to translate between arrays of numbers and this BSON Binary specification.
+
+#### Data Types
+
+Each vector can take one of multiple data types (dtypes). The following table lists the dtypes implemented.
+
+| Vector data type | Alias      | Bits per vector element | [Arrow Data Type](https://arrow.apache.org/docs/cpp/api/datatype.html) (for illustration) |
+| ---------------- | ---------- | ----------------------- | ----------------------------------------------------------------------------------------- |
+| `0x03`           | INT8       | 8                       | INT8                                                                                      |
+| `0x27`           | FLOAT32    | 32                      | FLOAT                                                                                     |
+| `0x10`           | PACKED_BIT | 1     `*`               | BOOL                                                                                      |
+
+`*` A Binary Quantized (PACKED_BIT) Vector is a vector of 0s and 1s (bits), but it is represented in memory as a list of
+integers in \[0, 255\]. So, for example, the vector `[0, 255]` would be shorthand for the 16-bit vector
+`[0,0,0,0,0,0,0,0,1,1,1,1,1,1,1,1]`. The idea is that each number (a uint8) can be stored as a single byte. Of course,
+some languages, Python for one, do not have an uint8 type, so must be represented as an int in memory, but not on disk.
+
+The authors are well-aware of the inherent ambiguity here, and alternatives. This is a market-standard, unfortunately.
+Change is inevitable.
+
+#### Byte padding
+
+As not all data types have a bit length equal to a multiple of 8, and hence do not fit squarely into a certain number of
+bytes, a second piece of metadata, the "padding" is included. This instructs the driver of the number of bits in the
+final byte that are to be ignored.
```

Comment:
Is it byte padding or bit padding?

I am not fully grokking what it means to ignore, from the example below:

```
// For example, a vector `[6, 7]` ... with a padding of `3`
0x10          0x03         0x06         0x07
00001 0000    0000 0011    0000 0110    0000 0111
```

Wouldn't the last 3 bits of the final byte here be "ignored" making the vector a total of 13-bits long ignoring the last three "111"?

Reply:
I've chosen a bad example, as it will thrown away the only values with 1s. I'll change it.

Bit padding or byte padding. Could be either. It's padding to make sureI hope that the description itself explains it. 

* f50b1cce -  Further improvements to PACKED_BIT with padding example.

Reply:
I would change it to bit padding because it is the bits that are doing the padding, it is the bytes that are being padded, the verb tense kinda has an implication. 

But I agree the prose are more important, should we add here that the bits to be ignored are the least significant ones? The example demonstrates it but probably should be spelled out here.

Reply:
RE: bit or byte padding - we use bits to pack the byte. Either word makes sense and requires explanation, so I've left as-is.